scipy/stats/_censored_data.py

```diff
@@ -0,0 +1,224 @@
+import numpy as np
+
+
+def _validate_lower_upper(lower, upper):
+    if lower.ndim != 1 or upper.ndim != 1:
+        raise ValueError('The input arrays must be one-dimensional.')
+    if np.shape(lower) != np.shape(upper):
+        raise ValueError('lower and upper must have the same length.')
+    if np.isnan(lower).any() or np.isnan(upper).any():
+        raise ValueError('lower and upper must not contain nan.')
+    if (np.isinf(lower) & np.isinf(upper)).any():
+        raise ValueError('lower and upper must not both be infinite.')
+    if (lower > upper).any():
+        raise ValueError('Elements of lower must not be greater than the '
+                         'corresponding elements of upper.')
+
+
+def _validate_x_censored(x, censored):
+    x = np.asarray(x)
+    if x.ndim != 1:
+        raise ValueError('x must be one-dimensional.')
+    censored = np.asarray(censored)
+    if censored.ndim != 1:
+        raise ValueError('censored must be one-dimensional')
+    if (~np.isfinite(x)).any():
+        raise ValueError('x must not contain nan or inf.')
+    if censored.size != x.size:
+        raise ValueError('x and censored must have the same length.')
+    return x, censored.astype(bool)
+
+
+class CensoredData:
+    """
+    Instances of this class represent censored data.
+
+    Instances may be passed to the ``fit`` method of continuous
+    univariate SciPy distributions for maximum likelihood estimation.
+
+    Left-, right-, and interval-censored data can be represented.
+
+    For interval-censored data, pass the lower and upper limits of
+    the intervals as one-dimensional arrays to the ``CensoredData``
+    constuctor.  By using the values `np.inf` and `-np.inf`, left-
+    and right-censored data may also be created with the constructor.
+
+    For convenience, the class methods ``left_censored`` and
+    ``right_censored`` are provided to create a ``CensoredData``
+    instance from a single one-dimensional array of measurements
+    and a corresponding boolean array to indicate which measurements
+    are censored.
+
+    Parameters
+    ----------
+    lower, upper : array_like
+        The data, a possibly mixed collection of censored and uncensored
+        measurements.  The arrays must be one-dimensional and have the same
+        length.  The values are interpreted as follows:
+
+        * ``lower[k] == upper[k]``: the common value is an uncensored
+          measurement.
+        * ``upper[k] == inf``: the measurement is right-censored; the
+          value in ``lower[k]`` is the lower bound for the true unknown
+          value.
+        * ``lower[k] == -inf``: the measurement is left-censored; the value
+          in ``upper[k]`` is the upper bound for the true unknown value.
+        * ``lower[k] < upper[k]``, and both values are finite: the
+          measurement is interval-censored; the true unknown value is
+          between ``lower[k]`` and ``upper[k]``.
```

Comment:
This description is still quite confusing. Compare with text from https://en.wikipedia.org/wiki/Censoring_(statistics):

- Left censoring – a data point is below a certain value but it is unknown by how much.
- Interval censoring – a data point is somewhere on an interval between two values.
- Right censoring – a data point is above a certain value but it is unknown by how much.

This is way clearer, consider adoption that verbatim above the Parameters description.

Reply:
It's not just the description. Left censoring has an _upper bound only_, but with this API you have to specify it as `upper[k]  = inf`, `lower[k] = upper_bound`.